### ISSUE TITLE
Velero Backup stuck in BackupStorageLocations is valid loop

### ISSUE THREAD

**Issue by RobertSkawinski** [vmware-tanzu/velero#6546]
**What steps did you take and what happened:**
```
velero install     --provider azure     --plugins velero/velero-plugin-for-microsoft-azure:v1.7.1     --bucket container-velero-test     --secret-file ./credentials-velero-test     --velero-pod-cpu-request 1000m     --velero-pod-mem-request 1Gi     --velero-pod-cpu-limit 4000m     --velero-pod-mem-limit 8Gi     --backup-location-config resourceGroup=rg-velero-test,storageAccount=velerotest     --snapshot-location-config apiTimeout=5m,resourceGroup=rg-velero-test
kubectl logs deployment/velero -n velero
kubectl label --all --dry-run=true sc velero.io/exclude-from-backup=true
kubectl label -n test--dry-run=true svc/aks-egress velero.io/exclude-from-backup=true
velero backup create fullbkptest20230725 --wait
```

**What did you expect to happen:**
Backup finished

**The following information will help us better understand what's going on**:
It seems like velero is stucking in a loop
```
time="2023-07-25T15:11:03Z" level=info msg="Using storage account key: true" backup-storage-location=velero/default cmd=/plugins/velero-plugin-for-microsoft-azure controller=backup-storage-location logSource="/go/src/velero-plugin-for-microsoft-azure/velero-plugin-for-microsoft-azure/object_store.go:364" pluginName=velero-plugin-for-microsoft-azure
time="2023-07-25T15:11:03Z" level=info msg="Validating BackupStorageLocation" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:152"
time="2023-07-25T15:11:03Z" level=info msg="BackupStorageLocations is valid, marking as available" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:137"
```

_If you are using velero v1.7.0+:_  
[bundle-2023-07-25-17-24-18.tar.gz](https://github.com/vmware-tanzu/velero/files/12162939/bundle-2023-07-25-17-24-18.tar.gz)

**Environment:**
- Velero version 1.11.1: 
- Velero features (use `velero client config get features`): features: <NOT SET>
- Kubernetes version (use `kubectl version`):
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.6", GitCommit:"ff2c119726cc1f8926fb0585c74b25921e866a28", GitTreeState:"clean", BuildDate:"2023-01-18T19:22:09Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.6", GitCommit:"94c50547e633f1db5d4c56b2b305670e14987d59", GitTreeState:"clean", BuildDate:"2023-06-12T18:46:30Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
- Kubernetes installer & version: 1.22.11
- Cloud provider or hardware configuration: azure-k8s
- OS (e.g. from `/etc/os-release`): Ubuntu 18.04

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by Lyndon-Li** [comment]
@RobertSkawinski 
The `Validating BackupStorageLocation` logs are expected, the real problem is here:

```
time="2023-07-25T14:59:01Z" level=info msg="Updating backup's final status" backuprequest=velero/fullbkpvvtt20230725 controller=backup logSource="pkg/controller/backup_controller.go:309"
time="2023-07-25T14:59:01Z" level=error msg="error updating backup's final status" backuprequest=velero/fullbkpvvtt20230725 controller=backup error="Backup.velero.io \"fullbkpvvtt20230725\" is invalid: status.phase: Unsupported value: \"Finalizing\": supported values: \"New\", \"FailedValidation\", \"InProgress\", \"Completed\", \"PartiallyFailed\", \"Failed\", \"Deleting\"" logSource="pkg/controller/backup_controller.go:311"
```

The backup is trying to update the status to "Finalizing", however, the backup CRD doesn't support this status.
I suspect you are using an old Velero client with a newer Velero server, you should use only the client that matches to the server. Specifically, for your case, you should use Velero 1.11.1 client to install Velero.

**Comment by RobertSkawinski** [comment]
Hi @Lyndon-Li,
thank you for this hint! - I was able to fix it by removing all old CRD's.
Seems like this k8s Cluster had already an old velero setup, which was not in sync with the new velero client and velero install did not update the old CRD's.

With this command I was able to generate all delete commands for velero CRD's:
`for i in $(kubectl get crd | grep velero | awk '{print $1}'); do echo kubectl delete crd $i; done`

After executing them and installing velero again ... everything worked as expected.

Thank you @Lyndon-Li 

Issue solved
